### DISCUSSION TITLE
How to get snapshot CRs populated on a fresh k8s cluster

### DISCUSSION THREAD

**Discussion by bfeher-kx** [k8up-io/k8up#874] [Q&A] [answered]
Hi!

I wonder if there is a simple way to have the snapshots CR populated based on the restic repo.

I seen that if I start a backup its gets fetched and populated, even found which module is doing it.

https://github.com/k8up-io/k8up/blob/master/restic/kubernetes/snapshots.go

Is there a way to call this directly?

**Comment by Kidswiss** [comment]
Hi @bfeher-kx 

Currently the snapshots only get syncrhonized after the backup with the code snipped you linked. It can't be called directly.

I'm also wondering why you'd need that. Could you please elaborate your use-case? Then I may be able to understand your need better.

**Reply by bfeher-kx** [reply]
We would like to use K8up to backup/restore data between different deployments. We are developing an operator and that would rely on the snapshot CRs on the fresh target K8s deployment

**Reply by bfeher-kx** [reply]
Hi @Kidswiss 

Any idea?

**Reply by Kidswiss** [reply]
Are those deployments on the same cluster, but in another namespace? In that case your operator could copy the snapshot CRs and the necessary credentials to the new namespace? With the snapshot CRs and the credentials you have everything needed to restore the snapshots.

**Reply by bfeher-kx** [reply]
Hi @Kidswiss 

The target would be a new Kubernetes cluster. We were thinking about to export the CRs and apply it on the target, but some metadata would be different then.

**Reply by Kidswiss** [reply, marked as answer]
@bfeher-kx 

The metadata doesn't really matter to k8up for restores. The important parts are in the `spec` field. Currently we also just copy the snapshots around if we need a restore in a new namespace/cluster.
